src/cpp/fastdds/publisher/DataWriter.cpp

```diff
@@ -48,6 +48,24 @@ DataWriter::~DataWriter()
 {
 }
 
+ReturnCode_t DataWriter::enable()
+{
+    if (enable_)
+    {
+        return ReturnCode_t::RETCODE_OK;
+    }
+
+    if (false == impl_->get_publisher()->is_enabled())
+    {
+        return ReturnCode_t::RETCODE_PRECONDITION_NOT_MET;
+    }
+
+    enable_ = true;
```

Comment:
This should not be necessary, we are setting the value after enabling the implementation.